fuse/data/tokenizers/modular_tokenizer/modular_tokenizer.py

```diff
@@ -1148,6 +1149,9 @@ def decode(self, ids: Iterable, skip_special_tokens: Optional[bool] = False) ->
         Returns:
             str: _description_
         """
+        if isinstance(ids, Tensor):
+            # Tokens in 'self.decoder_dict' are integers, and not singletons
```

Comment:
singletons?

Reply:
Do we need also to move it to CPU before? in cases the tensor is on the GPU.

Reply:
Good question, so it's being done automatically:
> Returns the tensor as a (nested) list. For scalars, a standard Python number is returned, just like with [item()](https://pytorch.org/docs/stable/generated/torch.Tensor.item.html#torch.Tensor.item). Tensors are automatically moved to the CPU first if necessary.


https://pytorch.org/docs/stable/generated/torch.Tensor.tolist.html

Reply:
> singletons?

Meaning a tensor with a single item in it. Will make it clearer